metric/system/process/process_common.go

```diff
@@ -33,9 +34,43 @@ import (
 	sysinfo "github.com/elastic/go-sysinfo"
 )
 
-// ProcsMap is a map where the keys are the names of processes and the value is the Process with that name
+//ProcsMap is aconvinence wrapper type
 type ProcsMap map[int]ProcState
 
+// ProcsTrack is a map where the keys are the names of processes and the value is the Process with that name
+//type ProcsTrack map[int]ProcState
+type ProcsTrack struct {
+	pids ProcsMap
+	mut  *sync.Mutex
```

Comment:
Why is this a pointer? I'm not sure it needs to be one. The zero value is safe for use.